### PR TITLE
USWDS-Site - Accessibility checklists: Migrate template to layouts

### LINKED ISSUE #2503: Move `accessibility-tests/index.html` to `_layouts` 1. Benefit being you can define layout in MD file.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

**Implemented `layout` pattern on accessibility checklist pages.** Now we can define the `accessibility-test` layout in frontmatter.

## Breaking change

This is not a breaking change.

## Related issue

Closes #2503

Task 1 of issue https://github.com/uswds/uswds-site/issues/2548

## Related pull requests

No changelog, not a user face change.

## Preview link

- [Accordion accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/accordion/accessibility-tests/)
- [Button accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/button/accessibility-tests/)
- [Icon accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/icon/accessibility-tests/)
- [Link accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/link/accessibility-tests/)
- [Table accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/table/accessibility-tests/)
- [Typography accessibility page →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-accessibility-checklist-layout/components/typography/accessibility-tests/)

## Problem statement

Accessibility test pages are currently using the accessibility test `include` instead of following the `layout` pattern used by other pages on the site.

## Solution

Create `accessibility-test` layout and define the layout in `{component}/accessibility.md` pages.

> [!NOTE]
I tried creating defaults in `_config.yml` but was unable to due to the limitation of needed to use pre-defined types or a collection. I was unable to create a collection because it uses the `_component` file path.
> 

## Major changes

- Created `accessibility-test` layout
- Use this layout in `accessibility.md` pages instead of `include`

## Testing and review

1. Visit accordion, button, icon, link, table, and typography accessibility pages
2. Confirm there is no visual regression
3. Confirm links on pages work as expected
4. Inspect layout usage and confirm it matches existing patterns.

### Testing checklist

- [ ]  No visual regression in all three component accessibility pages
- [ ]  Links function as expected
- [ ]  Layout pattern is correct
- [ ] Confirm markdown to html changes do not break content

### CODE REVIEW

**Comment by mahoneycm** [comment]
## Typography accessibility guidance button

Just a heads up, it looks like the typography component page is missing the accessibility guidance callout and link found on other pages.

image 

I believe this is because the typography page front matter is missing a components attribute since it isn't a component, and that's what the include uses to include the section:

```md
<!-- component-guidance.html -->
{% assign componentName = page.component.name | default: page.title | downcase %}
{% assign componentKey = componentName | slugify %}
{% assign settingsKey = componentKey %}
{% assign checklistData = site.data.accessibility-tests[componentKey] %}

...

{% if checklistData %}
{% include accessibility-tests/a11y-note.html %}
{% endif %}
```

Would we want to address this?

CC: @amyleadem  @mejiaj

**Comment by amyleadem** [comment]
@mahoneycm I think it should be addressed. I've opened up #2593 to address.

File: _includes/accessibility-tests/checklist-feedback.html
Comment (mejiaj) [inline _includes/accessibility-tests/checklist-feedback.html:1]: This file is currently included in another include.

https://github.com/uswds/uswds-site/blob/57f9c0e44c06e0c5ddc3bcb49d4a66337e3859ca/_includes/accessibility-tests/index.html#L53-L54


Is it more appropriate to move the include to the layout instead `_layouts/accessibility-test.html`?

https://github.com/uswds/uswds-site/blob/57f9c0e44c06e0c5ddc3bcb49d4a66337e3859ca/_layouts/accessibility-test.html#L1-L5

File: _includes/accessibility-tests/checklist-intros.html
Comment (mejiaj) [inline _includes/accessibility-tests/checklist-intros.html:1]: ## Todo

It seems odd that `title_plural` is used as the default when it's only an exception when `screen reader` is the value.

**Proposal**

```liquid
{% assign title = title | downcase %}

{% if title == "Screen reader" %}
  {% assign title = "screen readers" %}
{% endif %}

<p class="border-top padding-top-1 border-gray-10">
  USWDS performed these tests to check the {{ component_name }}'s accessibility and usability for {{ title }}.
  For help performing your own {{ title }} tests, check out the resources in the expandable section below.
</p>

<div class="usa-accordion site-checklist-accordion">
  <h4 class="usa-accordion__heading">
    <button type="button" class="usa-accordion__button" aria-controls="{{ accordion_id }}">
      How to test with {{ title }}
    </button>
  </h4>
```

File: _includes/accessibility-tests/jump-links.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## {{ component_name | capitalize }} accessibility checklist
+<h2>{{ component_name | capitalize }} accessibility checklist</h2>
```
Comment (mejiaj) [inline _includes/accessibility-tests/jump-links.html:1]:
```suggestion
<!-- TODO: Reduce repetition in this template. -->
<h2>{{ component_name | capitalize }} accessibility checklist</h2>
```

File: _layouts/accessibility-test.html
Comment (mejiaj) [inline _layouts/accessibility-test.html:1]: @amyleadem @mahoneycm does it make sense to move the markup from `index.html` here?

**Review by mejiaj — COMMENTED** [review]
Looks good, thanks for creating this PR.

I've added some comments for discussion/future improvement. Can we discuss briefly in this PR? Anything that's out of scope; let's create issues for it if y'all agree with the proposals.

File: _layouts/accessibility-test.html
Comment (mahoneycm) [reply _layouts/accessibility-test.html:1]: Resolved in [849a270](https://github.com/uswds/uswds-site/pull/2502/commits/849a270e4dc914a6fc364bb472dc0184bd948867)

File: _includes/accessibility-tests/checklist-feedback.html
Comment (mahoneycm) [reply _includes/accessibility-tests/checklist-feedback.html:1]: Resolved in [849a270](https://github.com/uswds/uswds-site/pull/2502/commits/849a270e4dc914a6fc364bb472dc0184bd948867)

**Review by mejiaj — APPROVED** [review]
@mahoneycm can you link the new issues from TODO's and resolve the comments before FFR?

File: _includes/accessibility-tests/jump-links.html (hunk 1 above)
Comment (amyleadem) [reply _includes/accessibility-tests/jump-links.html:2]: Update: I committed this TODO comment and opened issue #2599 for a follow-on effort.

File: _includes/accessibility-tests/checklist-intros.html
Comment (amyleadem) [reply _includes/accessibility-tests/checklist-intros.html:1]: I've added this as a low-effort task in issue #2438

**Review by thisisdano — APPROVED** [review]

**Comment by thisisdano** [comment]
LGTM!

**Comment by thisisdano** [comment]
Merging over pa11y crash :(

**Pull request merged**